runtime/onert/core/src/compiler/pass/PermutationInsertionPass.cc

```diff
@@ -129,14 +129,10 @@ ir::OperationIndex PermutationInsertionPass::insertPermute(const ir::OperandInde
 
   // Find Permute information
   auto input_backend = operand_li_map.getRawPtr(operand_index)->def_backends().getOnlyElement();
-  auto output_backend = backend;
   const backend::Backend *permute_node_backend = compiler::BackendManager::get().getBuiltin();
   assert(permute_node_backend->config()->id() == onert::backend::builtin::Config::ID);
+  assert(input_backend != backend);
```

Comment:
(Q) Why should not those backends be the same?

Reply:
`PermutationInsertionPass` was introduced for
1) Layout change between use-def
2) Backend change between use-def

Now 1) is impossible. So `insertPermute` is called when backend is changed.